api/v1alpha1/shared_types.go

```diff
@@ -275,3 +276,35 @@ const (
 	// https://github.com/google/re2/wiki/Syntax.
 	StringMatchRegularExpression StringMatchType = "RegularExpression"
 )
+
+// KubernetesHorizontalPodAutoscalerSpec defines Kubernetes Horizontal Pod Autoscaler settings of Envoy Proxy Deployment
+// See k8s.io.autoscaling.v2.HorizontalPodAutoScalerSpec
+type KubernetesHorizontalPodAutoscalerSpec struct {
+	// minReplicas is the lower limit for the number of replicas to which the autoscaler
+	// can scale down. It defaults to 1 replica.
+	//
+	// +optional
+	MinReplicas *int32 `json:"minReplicas,omitempty"`
+
+	// maxReplicas is the upper limit for the number of replicas to which the autoscaler can scale up.
+	// It cannot be less that minReplicas. It defaults to 1 replica.
+	//
+	// +optional
+	MaxReplicas *int32 `json:"maxReplicas,omitempty"`
```

Comment:
Should the MaxReplicas be mandatory?

Reply:
what about both `minReplicas` and `maxReplicas` be mandatory?

Reply:
I have no strong opinion on `minReplicas`, but at least `maxReplicas` should be mandatory.

Reply:
i have no concern, so let me update it